### PR TITLE
unit 4 assessment🖊️

### PR DESCRIPTION
**Pull request by smorgannicole**
# Submitting Unit 4 Assessment

### ✋ Reminder

Do NOT merge your assessment!

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [😤] React Challenge
- [😤] Interview Practice Questions
- [😤] Looking Ahead Questions

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: ruby.rb [hunk 1]
```diff
@@ -7,4 +7,26 @@
 # Add comments to your code explaining your approach
```
Comment (smorgannicole) [inline ruby.rb:8]: I just remembered I told myself I was going to go back and leave comments on the ruby testing part and I completely forgot so I will do that now... honestly I didn't look at the testing file except to get the method names, I was fully relying on the console to tell me what to do

// attr_accessor bc that's the getter/setter for the instance variables. And I think it allows the user to use dot notation when referencing the instance vars outside the class
// only model needs to be passed when instantiating a new instance of Bike bc all other instance vars have default values (and it makes sense for them to)
// current_speed method returns @speed bc that instance var holds the speed val
// pedal_faster method, when called, is intended to be passed a positive num that will be added to whatever val @speed is at while simultaneously resetting the val of @speed
// brake method works in a similar way; brake is intended to be passed a positive num that will be subtracted from the val of @speed. I did have to go back after the test continued to fail and refactor my code to account for vals larger than @speed and I used a ternary statement to do so
// the bike_info method is just regurgitating the current value of all the instance vars in a cutie little string interpolated statement

File: code-challenges.test.js [hunk 2]
```diff
@@ -13,26 +13,77 @@
...
 
 // b) Create the function that makes the test pass.
-
+const fibonacciSequencer = (arrLength) => {
+  // validating value is greater than 2
+  if (arrLength < 2) {
+    return
+  }
+  // what the arr would look like if arrLength was 2
+  let babyFib = [ 1, 1 ]
```
Comment (sjproctor) [inline code-challenges.test.js:42]: babyFib! love it.

File: code-challenges.test.js [hunk 3]
```diff
@@ -13,26 +13,77 @@
...
+    // bracket notation to extract the previous 2 values
+    // adding those two values together to create the next value in the sequence
+    // pushing the value into the babyFib arr
+    const previousVal = babyFib[i - 1]
+    const previousPreviousVal = babyFib[i - 2]
+    const nextFibVal = previousVal + previousPreviousVal
+    babyFib.push(nextFibVal)
+  }
+  return babyFib
+}
```
Comment (sjproctor) [inline code-challenges.test.js:57]: Nice work!

File: interview-questions.md [hunk 4]
```diff
@@ -10,40 +10,54 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 Researched answer:
 
-2. What does it mean to implement a TDD workflow?
+Differences
 
-Your answer:
+1. The general syntax of both
+2. JS is client-side whereas ruby is server-side
+3. JS is more scaleable than Ruby
+4. JS is asynchronous (executing tasks while waiting for other operations to complete) and ruby is synchronous
```
Comment (sjproctor) [inline interview-questions.md:30]: By default JavaScript is synchronous but there are lots of ways to add asynchronous functionality.

File: ruby.rb (hunk 1 above)
Comment (sjproctor) [reply ruby.rb:8]: Relying on the terminal is definitely the right approach!

**Review by sjproctor — APPROVED** [review]
Beautiful work Morgan! You are displaying a competent understanding of the material and a thorough approach to your coding and explanations. Keep up the good work.

**Pull request merged**